File: api/schemas/notes.py
from pydantic import BaseModel


class NoteOut(BaseModel):
    sender_name: str
    message: str
    timestamp: str  # ISO 8601
    is_manager_reply: bool
    avatar_data_uri: str | None = None


class NoteSubmitRequest(BaseModel):
    message: str
    as_operator: str | None = None

File: frontend/src/api/notes.ts
import { api } from './client'

export interface Note {
  sender_name: string
  message: string
  timestamp: string
  is_manager_reply: boolean
  avatar_data_uri: string | null
}

export const notesApi = {
  list: (asOperator?: string) =>
    api.get<Note[]>(asOperator ? `/notes?${new URLSearchParams({ as_operator: asOperator })}` : '/notes'),
  send: (message: string, as_operator?: string) => api.post<{ ok: boolean }>('/notes', { message, as_operator }),
}
